stdlib/src/sys/_assembly.mojo

```diff
@@ -40,7 +40,7 @@ fn inlined_assembly[
                 _type=None,
                 assembly = asm.value,
                 constraints = constraints.value,
-                hasSideEffects = __mlir_attr.unit,
+                has_side_effects = __mlir_attr.unit,
```

Comment:
hey thanks for this refactor! Unfortunately the attribute names for MLIR ops can't be changed as they need to be in sync with what the ops expect. Do you mind reverting all these `has_side_effects` name changes (in both files)? Thanks!

Reply:
+1 to what Billy said here: it's the reason internal CI is failing and we haven't been able to land this change yet.

Reply:
@msaelices if you rebase and take Billy's comments into account, we're happy to land this!

Reply:
Done. Thanks!